app/admin/Settings.scala

```diff
@@ -17,19 +24,59 @@ case class Settings(
 )
 
 object Settings {
-  def fromConfig(config: Config): Settings =
-    Settings(
-      Switches(
-        PaymentMethodsSwitch.fromConfig(config.getConfig("oneOff")),
-        PaymentMethodsSwitch.fromConfig(config.getConfig("recurring")),
-        experimentsFromConfig(config, "abtests"),
-        SwitchState.fromConfig(config, "optimize"),
-        SwitchState.fromConfig(config, "internationalSubscribePages")
-      )
+
+  def fromDiskOrS3(config: Config)(implicit s3: AmazonS3): Either[Throwable, Settings] =
+    for {
+      source <- AdminSettingsSource.fromConfig(config)
+      rawJson <- getRawJson(source).leftMap(err => new Error(s"Could not fetch settings from $source. $err"))
+      settings <- decodeJson(rawJson).leftMap(err => new Error(s"Could not decode settings JSON from $source. $err"))
+    } yield settings
```

Comment:
Might be worth logging whether we ended up reading these settings from S3 or from disk?